src/OptionsCompounderAave2.sol

```diff
@@ -271,57 +223,91 @@ contract OptionsCompounder is FlashLoanReceiverBase, Ownable {
-        console2.log("2.Balance of paymentToken: ", assetBalance);
-        console2.log("2.Amount to pay back: ", totalAmount);
 
-        if ((assetBalance - initialBalance) < totalAmount) {
+        if ((assetBalance - initialBalance) <= totalAmount) {
             revert OptionsCompounder__FlashloanNotProfitable(
                 (assetBalance - initialBalance),
                 totalAmount
             );
         }
-        _gain = (assetBalance - initialBalance) - totalAmount; // Question: still we shall use safe math ?
-        senderToBalance[sender] = _gain;
-        IERC20(asset).approve(address(LENDING_POOL), totalAmount);
+        /* Protected by statement above */
+        gainInPaymentToken = (assetBalance - initialBalance) - totalAmount;
+
+        /* Approve the underlying token to make swap */
+        IERC20(asset).approve(address(swapperSwaps), gainInPaymentToken);
+
+        /* Transfer gain to the sender */
+        swapperSwaps.swapBal(
+            asset,
+            wantToken,
+            gainInPaymentToken,
+            minAmountOutData,
+            BEETX_VAULT_OP
+        );
+
+        gainInWantToken = IERC20(wantToken).balanceOf(address(this));
 
-        return _gain;
+        /* Approve lending pool to spend borrowed tokens + premium */
+        IERC20(asset).approve(address(lendingPool), totalAmount);
+
+        /* Console log area - temporary */
+        console2.log(
+            "3.Balance of underlyingToken: ",
+            IERC20(underlyingToken).balanceOf(address(this))
+        );
+        console2.log("3.Balance of wantToken: ", gainInWantToken);
+        console2.log(
+            "3.Balance of paymentToken: ",
+            paymentToken.balanceOf(address(this))
+        );
+        console2.log("3.Amount paid back: ", totalAmount);
+
+        return gainInWantToken;
     }
 
     /***************************** Getters ***********************************/
-    function isStrategyAdded(address _strategy) public view returns (bool) {
-        address[] memory tmpStrategies = strategies;
-        bool strategyFound = false;
-        for (uint8 idx = 0; idx < tmpStrategies.length; idx++) {
-            if (tmpStrategies[idx] == _strategy) {
-                strategyFound = true;
-                break;
-            }
-        }
-        return strategyFound;
+    /* Temporary for testing - apy will be caluclated in vault */
+    function getLastGain() external view returns (uint256) {
+        return gain;
     }
 
-    function getNumberOfStrategiesAvailable() external view returns (uint256) {
-        return strategies.length;
+    function ADDRESSES_PROVIDER()
+        external
+        view
+        returns (ILendingPoolAddressesProvider)
+    {
+        return addressProvider;
     }
 
-    function getLastGain() external view returns (uint256) {
-        return gain;
+    function LENDING_POOL() external view returns (ILendingPool) {
```

Comment:
i think the caps here might actually be more confusing than it is helpful. also, why do we need to make lendingPool private, cant we just make it public and use its auto-gen getter

Reply:
IFlashLoanReceiver requires LENDING_POOL and ADDRESSES_PROVIDER in this form